// mp-android-chart/mpchart.android.js
/**
 *
 * @providesModule MPChart
 * @flow
 */
'use strict';

import { PropTypes } from 'react';
import { requireNativeComponent, View, TextInput } from 'react-native';

var MPChart ={
	name: 'MPChart',
	propTypes: {
		draw_bar_shadow: PropTypes.bool,
		draw_value_above_bar: PropTypes.bool,
		description: PropTypes.string,
		max_visible_value_count: PropTypes.number,
		pinch_zoom: PropTypes.bool,
		draw_grid_background: PropTypes.bool,
		x_axis_position: PropTypes.oneOf([
		  'TOP',
		  'BOTTOM',
		  'BOTH_SIDED',
		  'TOP_INSIDE',
		  'BOTTOM_INSIDE',
		]),
		x_axis_draw_grid_lines: PropTypes.bool,
		x_axis_space_between_labels:  PropTypes.number,
		y_axis_left_position: PropTypes.oneOf([
		  'OUTSIDE_CHART',
		  'INSIDE_CHART',
		]),
		y_axis_left_space_top: PropTypes.number,
		y_axis_left_min_value: PropTypes.number,
		y_axis_right_position: PropTypes.oneOf([
		  'OUTSIDE_CHART',
		  'INSIDE_CHART',
		]),
		y_axis_right_space_top: PropTypes.number,
		y_axis_right_min_value: PropTypes.number,
		legend_position: PropTypes.oneOf([
        'RIGHT_OF_CHART', 'RIGHT_OF_CHART_CENTER', 'RIGHT_OF_CHART_INSIDE',
        'LEFT_OF_CHART', 'LEFT_OF_CHART_CENTER', 'LEFT_OF_CHART_INSIDE',
        'BELOW_CHART_LEFT', 'BELOW_CHART_RIGHT', 'BELOW_CHART_CENTER',
        'ABOVE_CHART_LEFT', 'ABOVE_CHART_RIGHT', 'ABOVE_CHART_CENTER',
        'PIECHART_CENTER'
		]),
		legend_form:  PropTypes.oneOf([
		  'SQUARE', 'CIRCLE', 'LINE'
		]),
		legend_text_size: PropTypes.number,
		legend_x_entry_sapce: PropTypes.number,
		data_source: PropTypes.string,
		...View.propTypes,
	},
};

module.exports =  requireNativeComponent('MPAndroidBarChart', MPChart);
